### PR TITLE
Conversion of LongText, LongTextV2, and plaintext into QTI2.1

### PR DESCRIPTION
**Pull request by rakeshqdsa**
Following JIRA ticket has been merged
https://learnosity.atlassian.net/browse/WBL-6
https://learnosity.atlassian.net/browse/WBL-7
https://learnosity.atlassian.net/browse/WBL-8
https://learnosity.atlassian.net/browse/WBL-10
https://learnosity.atlassian.net/browse/WBL-13

### CODE REVIEW

**Comment by ttton** [comment]
Hi @rakeshqdsa, can you please give this pull request a short, clear and meaningful title? Could you also provide any context of the changes for the reviewer in the PR description. Thanks.

**Comment by rakeshqdsa** [comment]
> Hi @rakeshqdsa, can you please give this pull request a short, clear and meaningful title? Could you also provide any context of the changes for the reviewer in the PR description. Thanks.

Title changed
And all the Related JIRA ticket has been mentioned in description .

**Comment by ttton** [comment]
@rakeshqdsa just a note, if there's anything I mention on here that is already addressed in `release/v2.0.0-alpha`, just leave a comment to say `already fixed`. If not, and a fix is needed, let's create a commit on this PR first and then we'll deal with the integration branch after. Let me know if that makes sense.

File: src/Processors/QtiV2/Out/QuestionTypes/PlaintextMapper.php [hunk 1]
```diff
@@ -4,28 +4,45 @@
...
+
+        if (isset($metadata) && !empty($metadata->get_distractor_rationale())) {
+            $feedbackOptions['genral_feedback'] = $metadata->get_distractor_rationale();
+        }
 
         $interaction = new ExtendedTextInteraction($interactionIdentifier);
         $interaction->setLabel($interactionLabel);
         $interaction->setPrompt($this->convertStimulusForPrompt($question->get_stimulus()));
-        $interaction->setFormat(TextFormat::PLAIN);
+        $interaction->setFormat('plain');
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Processors/QtiV2/Out/QuestionTypes/PlaintextMapper.php]: Done

File: src/Processors/QtiV2/Out/QuestionTypes/PlaintextMapper.php [hunk 2]
```diff
@@ -4,28 +4,45 @@
...
         $interaction->setMinStrings(1);
-        $interaction->setMaxStrings(1);
+        
+        if (isset($questionData['max_length'])) {
+            $interaction->setExpectedLength($questionData['max_length']);
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Processors/QtiV2/Out/QuestionTypes/PlaintextMapper.php]: Correct we do not need to set expected length as this is for number of lines . We have changed this in all places .

File: src/Processors/QtiV2/Out/QuestionTypes/PlaintextMapper.php [hunk 3]
```diff
@@ -4,28 +4,45 @@
...
         /** @var plaintext $question */
         $question = $questionType;
+        $questionData = $question->to_array();
+
+        // $feedbackOptions is used to create feedback or modelfeedback elements in responseProcessing
+        $metadata = $question->get_metadata();
+        $feedbackOptions = [];
+
+        if (isset($metadata) && !empty($metadata->get_distractor_rationale())) {
+            $feedbackOptions['genral_feedback'] = $metadata->get_distractor_rationale();
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Processors/QtiV2/Out/QuestionTypes/PlaintextMapper.php]: Done

File: src/Processors/QtiV2/Out/QuestionTypes/LongtextV2Mapper.php [hunk 4]
```diff
@@ -14,18 +15,34 @@ public function convert(BaseQuestionType $questionType, $interactionIdentifier,
...
         $question = $questionType;
 
+        $questionData = $question->to_array();
+
+        // $feedbackOptions is used to create feedback or modelfeedback elements in responseProcessing
+        $metadata = $question->get_metadata();
+        $feedbackOptions = [];
+
+        if (isset($metadata) && !empty($metadata->get_distractor_rationale())) {
+            $feedbackOptions['genral_feedback'] = $metadata->get_distractor_rationale();
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Processors/QtiV2/Out/QuestionTypes/LongtextV2Mapper.php]: Done

File: src/Commands/ConvertToQtiCommand.php [hunk 5]
```diff
@@ -30,6 +30,13 @@ protected function configure()
                 'An output path where the QTI will be saved',
                 './data/output'
             )
+            ->addOption(
+                'format',
+                'f',
+                InputOption::VALUE_REQUIRED,
+                'Conversion format for LMS currenlty only supported canvas LMS (if you want to QTI conversion supported for canvas please provide -f=canvas default is qti)',
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Commands/ConvertToQtiCommand.php]: Done

**Pull request merged**